book/src/todo_3.md

```diff
@@ -0,0 +1,205 @@
+It also loads stylesheets automatically from resources as long as they are named [accordingly](https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/struct.Application.html#automatic-resources).
+
+Looking at our To-Do app we can see that the looks of its widgets changed.
+This is because the `Default` stylesheet provided by GTK has been replaced with the `Adwaita` stylesheet provided by Libadwaita.
+
+<div style="text-align:center"><img src="img/todo_change_4_5.png"/></div>
+
+Now, if we go to the settings and request a different style, our app will oblige.
+
+<div style="text-align:center">
+ <video autoplay muted loop>
+  <source src="vid/todo_dark.webm" type="video/webm">
+Your browser does not support the video tag.
+ </video>
+</div>
+
+
+## Start using Libadwaita widgets
+
+Of course Libadwaita is more than just a couple of stylesheets and a [`StyleManager`](https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/struct.StyleManager.html).
+But before we get to the interesting stuff, we make our lives easier in the future by replacing all occurrences of `gtk::prelude` and `gtk::subclass::prelude` with [`adw::prelude`](https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/prelude/index.html) and [`adw::subclass::prelude`](https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/subclass/prelude/index.html).
+This works because the `adw` preludes re-export the corresponding `gtk` preludes plus a couple of Libadwaita specific traits.
+
+In the remainder of this chapter we are going to follow a couple of patterns of GNOME's HIG.
+Let's start by adapting the header bar to be [HIG compliant](https://developer.gnome.org/hig/patterns/containers/header-bars.html).
+We do that by replacing [`gtk::ApplicationWindow`](../docs/gtk4/struct.ApplicationWindow.html) with [`adw::ApplicationWindow`](https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/struct.ApplicationWindow.html).
+We also use [`adw::HeaderBar`](https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/struct.HeaderBar.html) as title bar.
+Finally, we add [tooltips](https://developer.gnome.org/hig/patterns/feedback/tooltips.html).
+
+The second pattern we are going to follow are [boxed lists](https://developer.gnome.org/hig/patterns/containers/boxed-lists.html).
+The HIG does not require us to use them and there's a good reason for that: they are incompatible with recycling lists.
+This means they cannot be used with [list views](https://developer.gnome.org/hig/patterns/containers/list-column-views.html) and are therefore only appropriate for relatively small lists.
+
+> Try to add tasks programmatically and see how many of them you have to add until the UI noticeably slows down.
```

Comment:
We're talking more than a thousand rows until it slows down I think

Reply:
10 000 even on my machine.